Suggestion for new dynamic detachable joint plugin

AdarshKaran opened the pull request:
<!--
Please remove the appropriate section.
For example, if this is a new feature, remove all sections except for the "New feature" section

If this is your first time opening a PR, be sure to check the contribution guide:
https://gazebosim.org/docs/all/contributing
-->

# 🎉 New feature

Closes -

## Summary
Added ros - gz bridge support for the the `dynamic_detachable_joint` system discussed in Issue https://github.com/gazebosim/gz-sim/issues/2362


## Test it

Adding the plugin to robot model URDF/SDF
```
    <gazebo>
      <plugin filename="gz-sim-dynamic-detachable-joint-system"
            name="gz::sim::systems::DynamicDetachableJoint">
        <parent_link>robotiq_85_left_finger_tip_link</parent_link>
        <service_name>/payload/attach_detach</service_name>
        <output_topic>/child_state</output_topic>
        <attach_distance>0.25</attach_distance>
      </plugin>
    </gazebo>

```

GZ Service call command
```
# Attach
gz service -s /payload/attach_detach \
    --reqtype gz.msgs.AttachDetachRequest \
    --reptype gz.msgs.AttachDetachResponse \
    --timeout 3000 \
    --req 'child_model_name: "cube", 
           child_link_name: "link", 
           command: "attach"'

# Detach
gz service -s /payload/attach_detach \
    --reqtype gz.msgs.AttachDetachRequest \
    --reptype gz.msgs.AttachDetachResponse \
    --timeout 2000 \
    --req 'child_model_name: "cube", 
           child_link_name: "link", 
           command: "detach"'
```
ROS2 Service call command
```
# Attach
ros2 service call /payload/attach_detach \
    ros_gz_interfaces/srv/AttachDetach \
    "{child_model_name: 'cube', 
      child_link_name: 'link', 
      command: 'attach'}"

# Detach  
ros2 service call /payload/attach_detach \
    ros_gz_interfaces/srv/AttachDetach \
    "{child_model_name: 'cube', 
      child_link_name: 'link', 
      command: 'detach'}"
```

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers
- [] Was GenAI used to generate this PR? If so, make sure to add "Generated-by" to your commits. (See [this policy](https://osralliance.org/wp-content/uploads/2025/05/OSRF-Policy-on-the-Use-of-Generative-Tools-Generative-AI-in-Contributions.pdf) for more info.) 

Generated-by: Remove this if GenAI was not used.

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` and `Generated-by` messages.

AdarshKaran committed:
added support for detachable joint service

Signed-off-by: Adarsh Karan Kesavadas Prasanth <adarshkaran01@gmail.com>

AdarshKaran committed:
added support for detachable joint service

Signed-off-by: Adarsh Karan Kesavadas Prasanth <adarshkaran01@gmail.com>

ahcorde reviewed (changes requested):
Do you mind to add some test ?

AdarshKaran committed:
updated attach detach bridge service API

Signed-off-by: Adarsh Karan Kesavadas Prasanth <adarshkaran01@gmail.com>

AdarshKaran commented:
Hi @ahcorde, I’ve pushed updates for the new gz-msgs API and added a test:

  - AttachDetach.srv uses command constants instead of a free string.
  - Response maps to error_code/message from gz.msgs.Result.
  - Bridge factory also now maps AttachDetach to gz.msgs.AttachDetachRequest / gz.msgs.Result.
  - Added a factory registration test for AttachDetach.

AdarshKaran committed:
removed unused attach detach request header

Signed-off-by: Adarsh Karan Kesavadas Prasanth <adarshkaran01@gmail.com>
